pkg/controller/job/tracking_utils.go

```diff
@@ -151,3 +151,26 @@ func isFinishedPodWithTrackingFinalizer(pod *v1.Pod) bool {
 	}
 	return (pod.Status.Phase == v1.PodFailed || pod.Status.Phase == v1.PodSucceeded) && hasJobTrackingFinalizer(pod)
 }
+
+type jobUIDCache struct {
```

Comment:
The implementation looks ok, but I would prefer to use some built-in primitives. Can we use the store as for expectations or backoffStore? Or sync.Map maybe?